doc/md/jdk16-access-exceptions.md

```diff
@@ -0,0 +1,62 @@
+# JDK 16+ access exceptions
+
+### Problem symptoms 
+On JDK 16 and above you may encounter InaccessibleObjectException or IllegalAccessException in following known cases:
+1. _Some_ usages of `mockkStatic` on java standard library classes, e.g. 
+   ```kotlin
+   mockkStatic(Instant::class)
+   every { Instant.now() } returns Instant.parse("2022-09-15T09:31:26.919Z")
+   ```
+   ```
+   java.time.format.DateTimeParseException: Text '2022-09-15T09:31:26.919Z' could not be parsed:
+     Unable to make private static java.time.Instant java.time.Instant.create(long,int) accessible: 
+     module java.base does not "opens java.time" to unnamed module @4501b7af
+   ```
+
+2. Spying on java standard library classes, e.g.
+   ```kotlin
+   val socket = Socket()
+   val spy = spyk<Socket>(socket)
+   ```
+    Note: this includes creating `@SpyKBean` on spring data repository instances, because they are instances of `java.lang.reflect.Proxy` in fact.
+   ```
+   Error creating bean with name 'featureRepository': Post-processing of FactoryBean's singleton object failed;
+    nested exception is java.lang.IllegalAccessException: class io.mockk.impl.InternalPlatform 
+    cannot access a member of class java.lang.reflect.Proxy (in module java.base) with modifiers "protected"
+   ```
+
+### Problem cause
+JDK 16 enforced strong encapsulation of standard modules. In practice this means you (and your libraries) can no more get away with call like `privateMember.setAccessible(true)` to any of JDK classes.
```

Comment:
```suggestion
JDK 16 enforces strong encapsulation of standard modules. In practice this means libraries are forbidden from overriding accessibility modifiers in JDK classes (for example,  `privateMember.setAccessible(true)`).
```

Reply:
> In practice this means libraries are forbidden

AFAIK not only libraries. Do you want to make accent on mockk use-case?

Reply:
Good point, I've edited it. 

I think the project name should be styled as: MockK